File: frontend/src/types/session.ts
/** 会话 */
export interface Session {
  id: number;
  user_id: number;
  title: string;
  mode: "chat" | "research";
  created_at: string;
  updated_at: string;
  message_count: number;
}

/** 会话详情 */
export interface SessionDetail extends Session {
  messages: Message[];
}

/** 创建会话请求 */
export interface CreateSessionRequest {
  title: string;
  mode?: "chat" | "research";
}

/** 更新会话请求 */
export interface UpdateSessionRequest {
  title?: string;
}

/** 会话列表响应 */
export interface SessionListResponse {
  items: Session[];
  total: number;
  page: number;
  page_size: number;
}

/** 消息 */
export interface Message {
  id: number;
  session_id: number;
  role: "user" | "assistant" | "system";
  content: string;
  citations?: Citation[];
  meta_info?: Record<string, unknown>;
  created_at: string;
}

/** 引用 */
export interface Citation {
  id: string;
  index: number;
  url: string;
  title: string;
  snippet: string;
  source_type: "web" | "mcp" | "memory" | "document";
  favicon?: string;
}

/** 聊天请求 */
export interface ChatRequest {
  session_id?: number;
  message: string;
  stream?: boolean;
}

/** 聊天响应 */
export interface ChatResponse {
  message_id: number;
  content: string;
  citations?: Citation[];
  session_id: number;
}

/** SSE 事件类型 */
export interface ChatStreamEvent {
  event: "message" | "chunk" | "citations" | "error" | "done";
  data: Record<string, unknown>;
}

File: backend/app/schemas/session.py
"""会话相关 Pydantic Schema."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.schemas.message import MessageResponse


class SessionBase(BaseModel):
    """会话基础 Schema."""

    title: str = Field(..., min_length=1, max_length=255)
    mode: str = Field(default="chat", pattern="^(chat|research)$")


class SessionCreate(SessionBase):
    """创建会话请求 Schema."""

    pass


class SessionUpdate(BaseModel):
    """更新会话请求 Schema."""

    title: Optional[str] = Field(None, min_length=1, max_length=255)


class SessionResponse(SessionBase):
    """会话响应 Schema."""

    id: int
    user_id: int
    created_at: datetime
    updated_at: datetime
    message_count: int = 0

    model_config = {"from_attributes": True}


class SessionDetailResponse(SessionResponse):
    """会话详情响应（包含消息列表）."""

    messages: list[MessageResponse] = []


class SessionListResponse(BaseModel):
    """会话列表响应."""

    items: list[SessionResponse]
    total: int
    page: int
    page_size: int
